junit-runtime/src/main/scala/org/hamcrest/CoreMatchers.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * Ported from https://github.com/hamcrest/JavaHamcrest/
+ */
+package org.hamcrest
+
+import org.hamcrest.core._
+
+object CoreMatchers {
+//  def allOf[T](matchers: java.lang.Iterable[Matcher[T]]): Matcher[T] =
+//    AllOf.allOf(matchers)
+
+//  def allOf[T](matchers: Matcher[T]*): Matcher[T] =
+//    AllOf.allOf(matchers)
+
+//  def anyOf[T](matchers: java.lang.Iterable[Matcher[T]]): AnyOf[T] =
+//    AnyOf.anyOf(matchers)
+
+//  def anyOf[T](matchers: Matcher[T]*) : AnyOf = AnyOf.anyOf(matchers)
+
+//  def both[LHS](matcher: Matcher[LHS]):
+//      CombinableMatcher.CombinableBothMatcher[LHS] = {
+//    CombinableMatcher.both(matcher)
+//  }
+
+//  def either[LHS](matcher: Matcher[LHS]):
+//      CombinableMatcher.CombinableEitherMatcher[LHS] = {
+//    CombinableMatcher.either(matcher)
+//  }
+
+//  def describedAs[T](description: String, matcher: Matcher[T],
+//      values: AnyRef*): Matcher[T] = {
+//    DescribedAs.describedAs(description, matcher, values)
+//  }
+
+//  def everyItem[U](itemMatcher: Matcher[U]): Matcher[java.lang.Iterable[U]] =
+//    Every.everyItem(itemMatcher)
```

Comment:
Those are matchers that use reflection under the hood in the original implementation.